src/tokenizer_test.cc

```diff
@@ -298,6 +298,56 @@ TEST_F(TokenizerTest, StringStartingWithNum) {
   EXPECT_EQ("/ABC", next->AsString());
 }
 
+TEST_F(TokenizerTest, StringQuotedSingleLine) {
+  Tokenizer t("\"Hello world\"");
+  auto next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsString());
+  EXPECT_EQ("Hello world", next->AsString());
+
+  next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsEOS());
+}
+
+TEST_F(TokenizerTest, StringQuotedMultiLine) {
+  Tokenizer t("\"Hello\n\nworld\"");
+  auto next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsString());
+  EXPECT_EQ("Hello\n\nworld", next->AsString());
+
+  next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsEOS());
+}
+
+TEST_F(TokenizerTest, StringQuotedUnterminated) {
+  Tokenizer t("\"Hello\n\nworld");
+  auto next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsString());
+  EXPECT_EQ("Hello\n\nworld", next->AsString());
+
+  next = t.NextToken();
+  ASSERT_TRUE(next != nullptr);
+  EXPECT_TRUE(next->IsEOS());
+}
+
+TEST_F(TokenizerTest, StringQuotedEscapeSequences) {
+  Tokenizer t(R"("_\\_a\aa_b\bb_t\tt_n\nn_v\vv_f\ff_r\rr_")");
```

Comment:
Throw a \" into this one to make sure embedded quotes work?

Reply:
Done. Split into two changes for sanity's sake.